### DISCUSSION TITLE
Modifying Elements on ParentScreen in Javascript

### DISCUSSION THREAD

**Discussion by th3112** [iobroker-community-adapters/ioBroker.webui#343] [Q&A] [answered]
Hi,
this new adapter for designing own UI for iObroker is very useful. I am new to this and I like the way of using own HTML, CSS and Javascript code within the creation of own UIs.

But this brings me to my question: SetElementProperty allows to manipulate Elements on the ParentScreen. Is this also possible if I am using Javascript instead of Simple Script? If yes, how to do?

Thanks in advance.

BR Thomas

**Comment by jogibear9988** [comment]
for sure...

like this:

        /**
        * @param {BaseScreenViewerAndControl} instance
        */
        export function init(instance) {
            debugger;
            /** @type {BaseScreenViewerAndControl} */
            let parentSV = instance.getRootNode().host.getRootNode().host
            //bspw:
            parentSV._getDomElement('aaa');
            //bspw:
            parentSV._getDomElements('.aaaa')
        }

**Comment by jogibear9988** [comment]
Problem is, this is not very refactor friendly. Cause I use inside of the screenviewer 2 shadow doms (for some reason), you need to stack those 4 calls.
I look to add a method or property wich makes this easier, and wich I could change if I adjust some of this

**Comment by jogibear9988** [comment, marked as answer]
So, I now added 2 methods 2 "RUNTIME", see:

    let parentSV = RUNTIME.getParentScreen(instance);
    let parentSV2 = RUNTIME.findParent(instance, ScreenViewer, x => x.id == 'aa');

With the second one you cound find any element with the constructor in the second parameter and a predicate wich you get put in the 3rd (the predicate also could be empty)

**Comment by th3112** [comment]
Hi,many thanks for the great effort and the quick provision of a solution.BrThomasAm 21.08.2024 um 21:30 schrieb Jochen Kühner ***@***.***>:﻿
So, I now added 2 methods 2 "RUNTIME", see:
let parentSV = RUNTIME.getParentScreen(instance);
let parentSV2 = RUNTIME.findParent(instance, ScreenViewer, x => x.id == 'aa');

With the second one you cound find any element with the constructor in the second parameter and a predicate wich you get put in the 3rd (the predicate also could be empty)
